typing/printtyp.ml

```diff
@@ -1277,30 +1277,33 @@ let rec tree_of_type_decl id decl =
   in
   let (name, args) = type_defined decl in
   let constraints = tree_of_constraints params in
-  let ty, priv =
+  let ty, priv, unboxed =
     match decl.type_kind with
     | Type_abstract ->
         begin match ty_manifest with
-        | None -> (Otyp_abstract, Public)
+        | None -> (Otyp_abstract, Public, false)
         | Some ty ->
-            tree_of_typexp false ty, decl.type_private
+            tree_of_typexp false ty, decl.type_private, false
         end
-    | Type_variant cstrs ->
+    | Type_variant (cstrs, rep) ->
         tree_of_manifest (Otyp_sum (List.map tree_of_constructor cstrs)),
-        decl.type_private
-    | Type_record(lbls, _rep) ->
+        decl.type_private,
+        (rep = Variant_unboxed)
```

Comment:
I would prefer a pattern-matching as below. (I would even prefer a non-fragile pattern, but here I think it's a matter of taste and there are reasonable arguments for the fragile version.)

Reply:
Changing fragile equalities to non-fragile pattern matches I can get behind, but what's the advantage of a fragile pattern match over an equality check?

Reply:
The record case below has very similar code but uses a fragile pattern. I think the code would be more consistent if they both used the same approach (and records cannot use equality directly because it takes an argument), but this is nitpicking territory and you've already been very kind to add many robust matchings, so I'll stay quiet on this one. Thanks!